### PR TITLE
TASK: Make caching behavior opt-in via `X-FullPageCache-EnableFusionAutoconfiguration` http header

### LINKED ISSUE #14: Header for Cache-Control not set properly
The cache header being set is `cachecontrol`, but it should be `cache-control` (see https://developer.mozilla.org/de/docs/Web/HTTP/Headers/Cache-Control). This leads to the browser ignoring the cache header, like seen in the attached image. Current cache age exceeds the expected max-age because the `cachecontrol` header is ignored.

msedge_hhvPtC13gh

Niklas will open a PR to fix this issue in the near future.

### LINKED ISSUE #15: Make it possible (easier?) to disable full page cache for custom controllers
I don't see any obvious way of disabling the full page cache from a custom controller action that is called via `GET` without cookies or query parameters.

Unsetting a header on the response seems to be impossible (even with `replaceHttpResponse`) since it's merged.

The full page cache should consider a cache header from the action response like `Cache-Control: no-cache`. This would still result in misses for all requests. Better yet would be some kind of exclusion _before_  the middleware e.g. by adding URI path exclusions via settings.

### PR DESCRIPTION
**Pull request by mficzel**
The middleware extracting the cache lifetime tags etc from fusion was renamed to `FusionAutoconfigurationMiddleware`. The middleware now has to be enabled for each response by setting the header `X-FullPageCache-EnableFusionAutoconfiguration` which is done for `Neos.Neos:Page` automatically. 

This prevents the package from interfering with custom controllers and also enables custom controllers to control the caching behavior via the headers `X-FullPageCache-Enabled`, `X-FullPageCache-Lifetime` and `X-FullPageCache-Tags`. In cases where fusion caching is used by custom controllers the `X-FullPageCache-EnableFusionAutoconfiguration` can be set to benefit from the autoconfiguration.

The setting of the public `Cache-Control` headers is moved into the `RequestCacheMiddleware` which allows it to be used in cases where the `FusionAutoconfigurationMiddleware` is not active. Alongside this a typo in the `Cache-Control` header  was fixed.

Resolves: https://github.com/Flowpack/Flowpack.FullPageCache/issues/14
Resolves: https://github.com/Flowpack/Flowpack.FullPageCache/issues/15

### CODE REVIEW

**Comment by mficzel** [comment]
Drove some circles with this pr but now think that things are in the right place. Maybe @hlubek and @patricekaufmann can check wether this solves their issues aswell.

**Comment by hlubek** [comment]
@mficzel Cool! I'll have a look if it solves the issue I had.

File: README.md [hunk 1]
```diff
@@ -51,6 +51,26 @@ Flowpack:
...
 
+How it works
+------------
+
+The package defines two http middlewares:
+  
+- `RequestCacheMiddleware`: If a request is cacheable the cache is asked first and only if no response is found the 
+  request is passed down the middleware chain. The cache lifetime and tags are determined from the 
+  `X-FullPageCache-Enabled`, `X-FullPageCache-Lifetime` and `X-FullPageCache-Tags` that are set by upstream middlewares 
+  or controllers. Additionally the middleware adds `ETag` and `CacheControl` Headers taking the lifetime and setting
```
Comment (albe) [inline README.md]:
```suggestion
  or controllers. Additionally the middleware adds `ETag` and `Cache-Control` Headers taking the lifetime and setting
```

**Review by kitsunet — APPROVED** [review]
Looks good, thank you!

**Comment by paavo** [comment]
I hope this gets merged an published soon 🤞

**Review by kdambekalns — APPROVED** [review]
👀👌 by 📚 – Looks good by reading

**Pull request merged**